File: php/src/CopyCharacterCodes.php
<?php

namespace Vestaboard\Vbml;

/**
 * Port of copyCharacterCodes.ts
 */
class CopyCharacterCodes
{
    public static function copy(array $characters): array
    {
        return array_map(fn($row) => array_values($row), $characters);
    }
}

File: python/pyvbml/copy_character_codes.py
"""Copy character codes.

Port of Vestaboard/vbml/src/copyCharacterCodes.ts
"""

from __future__ import annotations


def copy_character_codes(characters: list[list[int]]) -> list[list[int]]:
    """Return a deep copy of the character codes array."""
    return [row[:] for row in characters]
